src/pxl_scripts/px/http_trace_id/script.pxl

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018- The Pixie Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+''' HTTP Trace Id Tracer
+
+This script traces all HTTP/HTTP2 data on the cluster that includes an Otel traceparent http header.
+In addition, it allows filtering by trace id so all of the spans contained within that trace can
+be viewed.
+'''
+import px
+
+
+def http_trace_data(start_time: str, trace_id_filter: str):
+
+    df = px.DataFrame(table='http_events', start_time=start_time)
+
+    # Add context.
+    df.node = df.ctx['node']
+    df.pid = px.upid_to_pid(df.upid)
+    df = add_source_dest_columns(df)
+    df = add_source_dest_links(df, start_time)
+
+    # Determine the trace_id from the traceparent http header and filter out
+    # requests that don't include a trace id.
+    df.traceparent_id = px.pluck(df.req_headers, 'traceparent')
+    df.trace_id = px.replace(
+        "[A-Za-z0-9]+\-([A-Za-z0-9]+)\-[A-Za-z0-9]+\-[A-Za-z0-9]+",  # noqa: W605
+        df.traceparent_id, "\\1")
```

Comment:
The traceparent header uses the following format ([otel docs](https://uptrace.dev/opentelemetry/opentelemetry-traceparent.html)):
```
# {version}-{trace_id}-{span_id}-{trace_flags}
traceparent: 00-80e1afed08e019fc1110464cfa66635c-7a085853722dc6d2-01
```

For this script, we want to filter by trace id since we want to display all the spans of a trace. The regex capture group is matching against the `{trace_id}` field from the example above.